src/main/scala/workflow/Estimator.scala

```diff
@@ -1,21 +1,34 @@
-package pipelines
+package workflow
 
 import java.io.Serializable
 
 import org.apache.spark.rdd.RDD
 
+import scala.reflect.ClassTag
+
 /**
  * An estimator has a `fit` method which takes an input and emits a [[Transformer]]
  * @tparam A The type of input this estimator (and the resulting Transformer) takes
  * @tparam B The output type of the node this estimator produces when being fit
  */
-abstract class Estimator[A, B] extends Serializable {
+abstract class Estimator[A, B] extends EstimatorNode  {
   /**
-   * An estimator has a `fit` method which emits a pipeline node.
+   * An estimator has a `fit` method which emits a [[Transformer]].
    * @param data Input data.
-   * @return A PipelineNode (Transformer) which can be called on new data.
+   * @return A [[Transformer]] which can be called on new data.
    */
-  def fit(data: RDD[A]): Transformer[A, B]
+  protected def fit(data: RDD[A]): Transformer[A, B]
+
+  private[workflow] final def fit(dependencies: Seq[RDD[_]]): TransformerNode[_] = fit(dependencies.head.asInstanceOf[RDD[A]])
```

Comment:
I'm less concerned about the ordering of seq mattering (as I'm just thinking of it as representing function parameters in an untyped way, where order still matters). We really do have to be careful because we're throwing all the types away, and I could be convinced of Transformer1-** and Estimator1-** being the right thing to do instead if we can find an alternative to how the Gather operator is currently done.